packages/maumau-server/src/controllers/game.ts

```diff
@@ -2,18 +2,22 @@ import { celebrate, Joi, Segments } from 'celebrate';
 import express, { Request, Response } from 'express';
 
 import { ClientState } from '../game/client-state-adapter';
-import { Action } from '../game/reducer';
+import { Action, State } from '../game/reducer';
 import { actionSchema } from '../game/validation';
-import GameSessionService from '../service/game-session';
+import GameSessionService, { Session } from '../service/game-session';
+import BotController from './bot';
 
 export default class GameController {
   public router = express.Router();
   public static basePath: string = '/game';
   private gameSessionService: GameSessionService;
+  private botController: BotController;
 
   constructor(args: { gameSessionService: GameSessionService }) {
     this.gameSessionService = args.gameSessionService;
+    this.botController = new BotController();
```

Comment:
Bin gerade noch nicht zu tief in der Thematik drin, aber ich glaube wir sollten den BotController lieber an den GameState dranhängen. Dann kann der BotController immer dann getriggert werden, wenn eine Action dispatched wird. Der BotController könnte schauen ob er für einen spieler einspringen muss und verzögert antworten. Ich glaube wenn wir den BotController hier definieren sind wir eine Ebene zu weit oben. Was meinst du?

Eine Idee wäre den `BotController` in der `dispatch` function im `GameState` zu triggern oder ihn in einer `ListenerFunction` zu implementieren.

`packages/maumau-server/src/game/game-state.ts`
```
 public dispatch(action: Action): State {
    this.isDispatching = false;

    try {
      this.isDispatching = true;
      this.state = reducer(this.state, action);
    } finally {
      this.isDispatching = false;
    }

    for (const listener of this.listeners) {
      listener(this.getState(), this.dispatch.bind(this)); <---- hier werden alle listener getriggert
    }

    // <--- Hier könnte alternativ der BotController angesprochen werden

    return this.state;
  }
```

Reply:
Ja macht voll Sinn. Dann braucht man auch weniger code. 👍

Reply:
Denke es ist besser als Listener aber dann müsste man den listener auch noch die komplette GameState Klasse übergeben. Denke es reicht aber vollkommen aus wenn der BotController unterhalb der Listener angesprochen wird.